Lib/functools.py

```diff
@@ -928,11 +928,14 @@ class singledispatchmethod:
     """
 
     def __init__(self, func):
+        import weakref # see comment in singledispatch method
```

Comment:
```suggestion
        import weakref # see comment in singledispatch function
```

Reply:
Thanks for the careful reviewing! I applied the suggestions